pkg/agent/apiserver/handlers/featuregates/handler.go

```diff
@@ -32,31 +33,33 @@ type Response struct {
 
 // HandleFunc returns the function which can handle queries issued by 'antctl get featuregates' command.
 // The handler function populates Antrea Agent feature gates information to the response.
-// For now, it will return all feature gates included in features.DefaultAntreaFeatureGates for agent
-// We need to exclude any new feature gates which is not consumed by agent in the future.
 func HandleFunc() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		var featureGates []Response
-		status, version := "", ""
 		for df := range features.DefaultAntreaFeatureGates {
-			if features.DefaultMutableFeatureGate.Enabled(df) {
-				status = "Enabled"
-			} else {
-				status = "Disabled"
+			if features.AgentGates.Has(df) {
+				featureGates = append(featureGates, Response{
+					Component: "agent",
+					Name:      string(df),
+					Status:    getStatus(features.DefaultFeatureGate.Enabled(df)),
+					Version:   string(features.DefaultAntreaFeatureGates[df].PreRelease),
+				})
 			}
-			version = string(features.DefaultAntreaFeatureGates[df].PreRelease)
-			featureGates = append(featureGates, Response{
-				Component: "agent",
-				Name:      string(df),
-				Status:    status,
-				Version:   version,
-			})
 		}
-
+		sort.SliceStable(featureGates, func(i, j int) bool {
```

Comment:
given that 2 FeatureGates cannot have the same name, there doesn't seem to be a reason to use `sort.SliceStable` here instead of `sort.Slice`

applies to other occurrences as well